src/components/ItemRecognition/ItemRecognition.js

```diff
@@ -14,10 +14,24 @@ class ItemRecognition extends Component {
   webcam = React.createRef();
   mobileNet = new MobileNet();
 
+  constructor(props) {
+    super(props);
+
+    if (navigator.online) {
+      this.model = new Model();
+      this.webcam = React.createRef();
+      this.mobileNet = new MobileNet();
+    }
+  }
+
   componentDidMount() {
     this.props.setPrediction(null, null);
-    this.model.load();
-    this.controllerDataset = new ControllerDataset();
+
+    if (navigator.onLine) {
+      this.model = new Model();
```

Comment:
If a new Model is created here can the duplicate of this statement be removed from the constructor?